In [1]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt

In [10]:
# Datasets are stored in the 'Formula1-DatasetFull/raw_data/' directory
directory = 'Formula1-DatasetFull/raw_data/'

list_datasets = ['drivers', 'driver_standings', 'qualifying', 'results', 'sprint_results', 'seasons']

for dataset_name in list_datasets:
    # Load the dataset into a DataFrame
    dataset = pd.read_csv(f'{directory}{dataset_name}.csv')
    # Get the number of observations
    num_observations = dataset.shape[0]
    # Print the result
    print(f"\nNumber of observations for {dataset_name}: {num_observations}")


Number of observations for drivers: 857

Number of observations for driver_standings: 33902

Number of observations for qualifying: 9575

Number of observations for results: 25840

Number of observations for sprint_results: 120

Number of observations for seasons: 74


In [21]:
# The "seasons" dataset is stored in the 'Formula1-DatasetFull/raw_data/' directory
directory = 'Formula1-DatasetFull/raw_data/'
# Load the "seasons" dataset into a DataFrame
seasons = pd.read_csv(f'{directory}seasons.csv')
# Get the unique years in the 'year' column
unique_years = seasons['year'].unique()
# Get the number of unique years
num_unique_years = len(unique_years)
# Sort the unique years in ascending order
sorted_unique_years = sorted(unique_years)
#Print Results
print(f"\nNumber of years covered in the 'seasons' dataset: {num_unique_years}")
print("\nSorted unique years in the 'seasons' dataset:")
for i in range(0, len(sorted_unique_years), 8):
    print(', '.join(map(str, sorted_unique_years[i:i+8])))



Number of years covered in the 'seasons' dataset: 74

Sorted unique years in the 'seasons' dataset:
1950, 1951, 1952, 1953, 1954, 1955, 1956, 1957
1958, 1959, 1960, 1961, 1962, 1963, 1964, 1965
1966, 1967, 1968, 1969, 1970, 1971, 1972, 1973
1974, 1975, 1976, 1977, 1978, 1979, 1980, 1981
1982, 1983, 1984, 1985, 1986, 1987, 1988, 1989
1990, 1991, 1992, 1993, 1994, 1995, 1996, 1997
1998, 1999, 2000, 2001, 2002, 2003, 2004, 2005
2006, 2007, 2008, 2009, 2010, 2011, 2012, 2013
2014, 2015, 2016, 2017, 2018, 2019, 2020, 2021
2022, 2023


<font size = "3">

We are using the drivers, driver_standings, qualifying, results, and sprint_results dataset tables. There are 857 observations in the drivers dataset, with each row representing a different driver (a unique driverId). There are 33902 observations in the driver_standings dataset, with each row representing a 
There are 9575 observations for the qualifying dataset, with each row representing 
There are 25840 observations in the results dataset, with each row representing 
There are 120 observations in the sprint_results dataset, with each row representing
There are 74 observations for the seasons dataset, with each row representing
The years of the dataset range from 1950 to 2023, a total of 74 years.